node/hyperbahn/index.js

```diff
@@ -91,10 +92,19 @@ function HyperbahnClient(options) {
         'Must pass in top level tchannel');
     assert(options.tchannel.tracer, 'Top channel must have trace enabled');
     assert(options.serviceName, 'must pass in serviceName');
-    assert(Array.isArray(options.hostPortList),
-        'Must pass in hostPortList');
+    if (Array.isArray(options.hostPortList)) {
+        self.hostPortList = options.hostPortList;
+    } else if (typeof options.hostPortFile === 'string') {
+        var hostContent = safeSyncRead(options.hostPortFile);
+        assert(!hostContent.error, 'Failed to read the hostPortFile');
```

Comment:
It's more about polling until the file exists. If we corrupt the file or oops puppet or anything bad happens I want processes that restarted during that period to keep trying.

I.e. if you have the file; cool story. If you dont; keep trying until we fix the files.